File: web/src/services/articleService.ts
import request from "../utils/request";

class ArticleService {
    public static getList(keyword: string, categoryId: string | null, tabIds: string | null, page: number = 1, pageSize: number = 20): Promise<any> {

        if (categoryId) {
            return request.get("/api/v1/Articles/List", {
                keyword,
                categoryId,
                page,
                pageSize,
                tabIds
            });
        } else {
            return request.get("/api/v1/Articles/List", {
                keyword,
                page,
                pageSize,
                tabIds
            });
        }
    }

    public static get(id: string): Promise<any> {
        return request.get(`/api/v1/Articles/${id}`);
    }

    public static getRanking() {
        return request.get("/api/v1/Articles/Ranking");
    }
}

export {
    ArticleService
}

File: Blog/src/web/Blog.Shared/Service/ArticleService.cs
using Blog.Contracts.Blogger;
using Masa.Utils.Models;

namespace Blog.Shared.Service;

public class ArticleService : BaseService, IArticleService
{
    public async Task<PaginatedListBase<GetArticleListDto>> GetListAsync(string? keyword, Guid? categoryId,
        string? tabIds,
        int page = 1, int pageSize = 20)
    {
        return await GetAsync<PaginatedListBase<GetArticleListDto>>("api/article/list?keyword=" + keyword +
                                                                    "&categoryId=" + categoryId + "&tabIds=" + tabIds +
                                                                    "&page=" + page + "&pageSize=" + pageSize);
    }

    public async Task<ArticleDto> GetAsync(int id)
    {
        return await GetAsync<ArticleDto>("api/article/" + id);
    }

    public Task<List<GetArticleListDto>> GetRankingAsync()
    {
        throw new NotImplementedException();
    }

    public Task CreateAsync(CreateArticleDto dto)
    {
        throw new NotImplementedException();
    }

    public Task DeleteAsync(int id)
    {
        throw new NotImplementedException();
    }

    public Task LikeAsync(int id)
    {
        throw new NotImplementedException();
    }
}